aggregator/list_test.go

```diff
@@ -57,23 +58,89 @@ func TestMetricListPushBack(t *testing.T) {
 }
 
 func TestMetricListClose(t *testing.T) {
-	l := newMetricList(testShard, time.Second, testOptions())
+	var (
+		registered   int
+		unregistered int
+	)
+	mockFlushManager := &mockFlushManager{
+		registerFn:   func(PeriodicFlusher) error { registered++; return nil },
+		unregisterFn: func(PeriodicFlusher) error { unregistered++; return nil },
+	}
+	opts := testOptions().SetFlushManager(mockFlushManager)
+	l := newMetricList(testShard, time.Second, opts)
 	l.RLock()
 	require.False(t, l.closed)
 	l.RUnlock()
+	require.Equal(t, 1, registered)
 
 	l.Close()
 	require.True(t, l.closed)
+	require.Equal(t, 1, unregistered)
 
 	// Close for a second time should have no impact.
 	l.Close()
 	require.True(t, l.closed)
+	require.Equal(t, 1, registered)
+	require.Equal(t, 1, unregistered)
+}
+
+func TestMetricListFlushWithCutoverCutoffNanos(t *testing.T) {
+	var (
+		now            = time.Unix(1234, 5678)
+		nowFn          = func() time.Time { return now }
+		beforeNanosRes int64
+		flushTypeRes   flushType
+	)
+	opts := testOptions().SetClockOptions(clock.NewOptions().SetNowFn(nowFn))
+	l := newMetricList(testShard, time.Second, opts)
+	l.flushBeforeFn = func(beforeNanos int64, flushType flushType) {
+		beforeNanosRes = beforeNanos
+		flushTypeRes = flushType
+	}
+
+	for _, input := range []struct {
```

Comment:
nit: perhaps move the declaration of `input` the before the for loop?

Reply:
Hm why? it's only scoped to the for loop.

Reply:
Just seems cleaner to me, feels kind of cluttered with it in the for loop. Just a nit though, feel free to disregard.

Reply:
Hm yeah, just looked through the go doc, seems moving it out of the loop is the recommended way (https://golang.org/doc/code.html#Testing), will do.